azext_iot/tests/test_iot_dps_int.py

```diff
@@ -68,9 +71,39 @@ def _cleanup_enrollments(self, dps, rg):
     )
 
 
+def _ensure_dps_hub_link(self, dps, rg, hub):
+    cli = embedded_cli.EmbeddedCLI()
+    hubs = cli.invoke(
+        "iot dps linked-hub list --dps-name {} -g {}".format(dps, rg)
+    ).as_json()
+    if not len(hubs) or not len(
+        list(
+            filter(
+                lambda linked_hub: linked_hub["name"]
+                == "{}.azure-devices.net".format(hub),
+                hubs,
+            )
+        )
+    ):
+        discovery = IotHubDiscovery(self.cmd_shell)
+        target_hub = discovery.get_target(hub, rg)
```

Comment:
`discovery.get_target` will/should throw (instead of return None) if the hub doesn't exist -- don't think you need to do the `if not target_hub` check.